src/components/Fund/FundDetails/FundDetailsContainer.js

```diff
@@ -53,94 +54,108 @@ const FundDetailsContainer = ({
-
-      if (fundTypeId) {
-        mutator.fundType.GET({
-          params: {
-            query: `id==${fundTypeId}`,
+  const fetchFundDetails = useCallback(
+    () => {
+      setIsLoading(true);
+      mutator.fund.GET().then(response => {
+        setCompositeFund(response);
+        const {
+          fund: {
+            ledgerId,
+            fundTypeId,
+            allocatedFromIds,
+            allocatedToIds,
           },
-        });
-      }
+        } = response;
 
-      if (ledgerId) {
-        mutator.ledger.GET({
-          params: {
-            query: `id==${ledgerId}`,
-          },
-        });
-      }
+        if (fundTypeId) {
+          mutator.fundType.GET({
+            params: {
+              query: `id==${fundTypeId}`,
+            },
+          }).then(fundType => setFundTypeName(fundType[0].name));
+        }
 
-      if (allocatedFromIds.length) {
-        mutator.allocatedFrom.GET({
-          params: {
-            query: allocatedFromIds.map(fundId => `id==${fundId}`).join(' or '),
-          },
-        });
-      }
+        if (ledgerId) {
+          mutator.ledger.GET({
+            params: {
+              query: `id==${ledgerId}`,
+            },
+          }).then(res => setLedger(res[0]));
+        }
 
-      if (allocatedToIds.length) {
-        mutator.allocatedTo.GET({
-          params: {
-            query: allocatedToIds.map(fundId => `id==${fundId}`).join(' or '),
-          },
-        });
-      }
-    });
+        if (allocatedFromIds.length) {
+          mutator.allocatedFrom.GET({
+            params: {
+              query: allocatedFromIds.map(fundId => `id==${fundId}`).join(' or '),
+            },
+          }).then(allocatedFrom => setAllocatedFromFunds(allocatedFrom.map(f => f.name).join(', ')));
+        }
+
+        if (allocatedToIds.length) {
+          mutator.allocatedTo.GET({
+            params: {
+              query: allocatedToIds.map(fundId => `id==${fundId}`).join(' or '),
+            },
+          }).then(allocatedTo => setAllocatedToFunds(allocatedTo.map(f => f.name).join(', ')));
+        }
+        setIsLoading(false);
+      });
 
-    mutator.group.GET();
-    mutator.budgets.GET();
+      setIsLoading(true);
+      mutator.budgets.GET().then(response => {
+        setBudgets(response);
+        setIsLoading(false);
+      });
+    },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [params.id]);
+    [params.id],
+  );
+
+  const fetchFund = useCallback(
+    () => {
+      if (fetchFundResource) {
+        setIsLoading(true);
+        mutator.fund.GET()
+          .then(response => {
+            setCompositeFund(response);
+            setFetchFundResource(false);
+            setIsLoading(false);
+          });
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [params.id, fetchFundResource],
+  );
+
+  useEffect(fetchFundDetails, [params.id]);
+
+  useEffect(fetchFund, [fetchFundResource]);
```

Comment:
hmm, interesting. `fetchFundResource` is an state property. It's only me who thinks that this is complicates the fetching logic flow?

Reply:
also it's very hard to say what the difference between `fetchFund` and `fetchFundDetails`